AdyenDropIn/Components/PreApplePayComponent.swift

```diff
@@ -65,24 +56,29 @@ internal final class PreApplePayComponent: PresentableComponent, FinalizableComp
     /// :nodoc:
     internal init(paymentMethod: ApplePayPaymentMethod,
                   apiContext: APIContext,
-                  payment: Payment,
                   configuration: Configuration,
                   applePayConfiguration: ApplePayComponent.Configuration) throws {
         self.apiContext = apiContext
-        self._payment = payment
         self.paymentMethod = paymentMethod
         self.configuration = configuration
-
+        self.payment = applePayConfiguration.applePayPayment.payment
         self.applePayComponent = try ApplePayComponent(paymentMethod: paymentMethod,
                                                        apiContext: apiContext,
-                                                       payment: payment,
                                                        configuration: applePayConfiguration)
         self.applePayComponent.delegate = self
     }
 
     /// :nodoc:
     internal func didFinalize(with success: Bool, completion: (() -> Void)?) {
-        applePayComponent.didFinalize(with: success, completion: completion)
+        applePayComponent.didFinalize(with: success, completion: { [weak self] in
+            let rootViewController = UIApplication.shared.keyWindow?.rootViewController
```

Comment:
lets not do this again (lets not grab the `UIApplication.shared.keyWindow`) in production code, lets make the `presentationDelegate` required in the initializer instead.

Reply:
It is not about that. I need to check if `PKPaymentViewController` is currently on a screen.
Do you know a better way?